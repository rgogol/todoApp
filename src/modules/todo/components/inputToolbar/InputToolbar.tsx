import React, { useState } from 'react';
import { View, Text, TextInput, TouchableOpacity, Keyboard } from 'react-native';
import { useDispatch } from 'react-redux';
import { addNewTask } from '../../redux/actions';
import InputStyles from './styles';

const InputToolbar = () => {
  const [task, setTask] = useState('');
  const dispatch = useDispatch();

  const { container, inputStyles, buttonStyles, buttonTextStyles } = InputStyles;

  const onChangeValue = (value: string) => setTask(value);

  const submitAction = () => {
    dispatch(addNewTask(task));
    setTask('');
    Keyboard.dismiss();
  };

  return (
    <View style={container}>
      <TextInput
        style={inputStyles}
        placeholder="Type new task"
        value={task}
        onChangeText={onChangeValue}
        testID="input"
      />
      <TouchableOpacity testID="add-new-button" style={buttonStyles} onPress={submitAction} disabled={task === ''}>
        <Text style={buttonTextStyles}>Add</Text>
      </TouchableOpacity>
    </View>
  );
};

export default InputToolbar;
