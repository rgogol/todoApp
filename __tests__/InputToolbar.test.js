import React from 'react';
import { shallow } from 'enzyme';
import InputToolbar from '../src/modules/todo/components/inputToolbar';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
}));

describe('InputToolbar', () => {
  const useState = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState');

  beforeEach(() => {
    useStateSpy.mockImplementation(init => [init, useState]);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  const wrapper = shallow(<InputToolbar />).dive();

  it('render as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('press add button with empty value', () => {
    const button = wrapper.find({ testID: 'add-new-button' });
    const submitAction = jest.fn();

    button.props().onPress();

    expect(submitAction).toHaveBeenCalledTimes(0);
  });
});
