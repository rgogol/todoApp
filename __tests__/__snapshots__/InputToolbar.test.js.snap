// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputToolbar render as expected 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "padding": 10,
    }
  }
>
  <Component
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Type new task"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderBottomLeftRadius": 5,
        "borderColor": "#750d37",
        "borderTopLeftRadius": 5,
        "borderWidth": 1,
        "flex": 1,
        "padding": 10,
      }
    }
    testID="input"
    underlineColorAndroid="transparent"
    value=""
  />
  <ForwardRef
    disabled={true}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#750d37",
        "borderBottomRightRadius": 5,
        "borderColor": "#750d37",
        "borderTopRightRadius": 5,
        "borderWidth": 1,
        "justifyContent": "center",
        "paddingHorizontal": 15,
        "paddingVertical": 10,
      }
    }
    testID="add-new-button"
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
        }
      }
    >
      Add
    </Text>
  </ForwardRef>
</View>
`;
